src/mongocxx/test/bulk_write.cpp

```diff
@@ -351,4 +352,15 @@ TEST_CASE("passing write operations to append calls corresponding C function", "
         REQUIRE(called);
     }
 }
+
+TEST_CASE("calling empty on a bulk write before and after appending", "[bulk_write]") {
+    instance::current();
+    mongocxx::client client{mongocxx::uri{}, test_util::add_test_server_api()};
+    auto bw = client["db"]["coll"].create_bulk_write();
+
+    REQUIRE(bw.empty());
+    bw.append(model::insert_one(make_document(kvp("_id", 1))));
+    bw.execute();
```

Comment:
It might be nice to have a check after the append and before executing, since what I understand as the main use case being to check if a bulk write is not empty before executing it.
```suggestion
    bw.append(model::insert_one(make_document(kvp("_id", 1))));
    REQUIRE_FALSE(bw.empty());
    bw.execute();
```